### PR TITLE
Fix/reference number unique

### PR DESCRIPTION
**Pull request by sam-drumm**
## What

When a staff member creates an application triggering CreateNewApplicationUseCase, no validation or checks on existance are performed on the email address. The email address (if supplied) forms the base of the reference number, so if the email address is the same the hash will be the same. 

It's not a requirment that the email is provided in this workflow, and leaving it blank will create a unique reference number. We could force a unique number always on this path but their are cases where a legitimate email address might be used, and that hash is used if a resident logs into see their application.

So this seemed like the most sensible solution - check if the email exists and 409 if it does. We also add validation to normalise email address - both on staff and resident paths. 

** Should the check be on both normalised and the inputed email? ** 

## Also
- I'll handle the 409 on the FE,
- This check will pick up only existing hashes that are normalised (i.e. Sam.Drumm@cool.com won't be matched)
- I've asked the team if adding their own staff email to an application represents a legitimate workflow, if so we'll have to handle that.

### CODE REVIEW

File: HousingRegisterApi/V1/Gateways/DynamoDbGateway.cs [hunk 1]
```diff
@@ -355,9 +355,19 @@ public Application GetApplicationById(Guid id)
 
         public Application GetIncompleteApplication(string email)
         {
-            string reference = _hashHelper.Generate(email).Substring(0, 10);
+            return GetApplicationsByEmail(email).FirstOrDefault();
+        }
+
+        public IEnumerable<Application> GetApplicationsByEmail(string email)
+        {
+            var normalizedEmail = EmailNormalizer.Normalize(email);
```
Comment (Duslerke) [inline HousingRegisterApi/V1/Gateways/DynamoDbGateway.cs]: This normalises the already normalised email.
It was normalised in the use case.

File: HousingRegisterApi/V1/Gateways/DynamoDbGateway.cs [hunk 2]
```diff
@@ -355,9 +355,19 @@ public Application GetApplicationById(Guid id)
...
+            if (normalizedEmail == null)
```
Comment (Duslerke) [inline HousingRegisterApi/V1/Gateways/DynamoDbGateway.cs]: And it was checked for null in the UC.

**Comment by Duslerke** [comment]
Sonar is failing due to too low code coverage. A few extra tests might bring it over the edge.

**Comment by sam-drumm** [comment]
@Duslerke test coverage gap is on DynamoDbGateway but a seperate ticket will look at that overall.

**Review by Duslerke — APPROVED** [review]
